Lib/typing.py

```diff
@@ -1268,7 +1283,7 @@ def __init_subclass__(cls, *args, **kwargs):
         if error:
             raise TypeError("Cannot inherit from plain Generic")
         if '__orig_bases__' in cls.__dict__:
-            tvars = _collect_type_vars(cls.__orig_bases__)
+            tvars = _collect_type_vars(cls.__orig_bases__, typevar_types=_TypeVarLike)
```

Comment:
What's the difference between passing `_TypeVarLike` and `(TypeVar, ParamSpec)`? The effect would seem to be the same (at least until we introduce yet another variant).

Reply:
Yep that was my intention. I was thinking if the variadic generic PEP gets approved, they would need another variant in 3.11. So the meaning is slightly different.

`TypeVarLike` means `Generic` accepts all TypeVars indiscriminately.

`(TypeVar, ParamSpec)` is *only* for those two.

Reply:
But the current draft of PEP 646 doesn’t allow Generic[Shape], it requires Generic[*Shape].

We really don’t know what the future will bring here, I think the subtlety is lost. Better add a comment somewhere explaining how to use this, than hope you’ve got this right.

Reply:
Hmm you're right, I just took a second look at PEP 646's current draft implementation, and I noticed it doesn't use ``TypeVarLike`` at all. So I decided to convert all checks for ``TypeVarLike`` to ``(TypeVar, ParamSpec)`` to be more explicit. Whatever I anticipate currently will probably all be overwritten in the next Big PEP anyways, so I'll make things more explicit so other people don't trip over this.